DNN Platform/Library/Services/Mail/CoreMailProvider.cs

```diff
@@ -0,0 +1,216 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+using DotNetNuke.Common.Utilities;
+using DotNetNuke.Entities.Host;
+using DotNetNuke.Entities.Portals;
+using System;
+using System.Net;
+using System.Net.Mail;
+using System.Text.RegularExpressions;
+using Localize = DotNetNuke.Services.Localization.Localization;
+namespace DotNetNuke.Services.Mail
+{
+    public class CoreMailProvider : MailProvider
+    {
+        private static readonly Regex SmtpServerRegex = new Regex("^[^:]+(:[0-9]{1,5})?$", RegexOptions.Compiled);
+        private static string ConvertToText(string sHTML)
+        {
+            var formattedHtml = HtmlUtils.FormatText(sHTML, true);
+            var styleLessHtml = HtmlUtils.RemoveInlineStyle(formattedHtml);
+            return HtmlUtils.StripTags(styleLessHtml, true);
+        }
+        public override string SendMail(MailInfo mailInfo, SMTPInfo smtpInfo = null)
+        {
+            // validate smtp server
+            if (smtpInfo == null || string.IsNullOrEmpty(smtpInfo.Server))
+            {
+                if (string.IsNullOrWhiteSpace(Host.SMTPServer))
+                {
+                    return "SMTP Server not configured";
+                }
+
+                smtpInfo = new SMTPInfo()
+                {
+                    Server = Host.SMTPServer,
+                    Authentication = Host.SMTPAuthentication,
+                    Username = Host.SMTPUsername,
+                    Password = Host.SMTPPassword,
+                    EnableSSL = Host.EnableSMTPSSL,
+                };
+            }
+
+            // translate semi-colon delimiters to commas as ASP.NET 2.0 does not support semi-colons
+            if (!string.IsNullOrEmpty(mailInfo.To))
+            {
+                mailInfo.To = mailInfo.To.Replace(";", ",");
+            }
+
+            if (!string.IsNullOrEmpty(mailInfo.CC))
+            {
+                mailInfo.To = mailInfo.To.Replace(";", ",");
+            }
```

Comment:
should this not be mailInfo.CC instead of mailInfo.To ?

Reply:
Good catch. Copy/Paste nightmare I'll resolve this in next commit.